src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolver.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.ComponentInterfaces;
+using Microsoft.VisualStudio.Debugger.FunctionResolution;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal abstract class FunctionResolver :
+        FunctionResolverBase<DkmProcess, DkmClrModuleInstance, FunctionResolutionRequest>,
+        IDkmRuntimeFunctionResolver,
+        IDkmModuleInstanceLoadNotification
+    {
+        void IDkmRuntimeFunctionResolver.EnableResolution(DkmRuntimeFunctionResolutionRequest request, DkmWorkList workList)
+        {
+            var signature = ParseSignature(request.FunctionName);
+            if (signature == null)
+            {
+                // PROTOTYPE(BP): Report error.
+                return;
+            }
+            EnableResolution(request.Process, new FunctionResolutionRequest(request, signature));
+        }
+
+        void IDkmModuleInstanceLoadNotification.OnModuleInstanceLoad(DkmModuleInstance moduleInstance, DkmWorkList workList, DkmEventDescriptorS eventDescriptor)
+        {
+            var module = moduleInstance as DkmClrModuleInstance;
+            if (module == null)
+            {
+                // Only interested in managed modules.
```

Comment:
Once you add the filter to the vsdconfigxml, you should never get here.  Assert here?